### PR TITLE
PE-5608: feat(logging)

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
- enhance error handling on encryption and decryption
- filter `TransactionDecryptionException` for buggy versions

--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/13f4smmp248dg

### CODE REVIEW

File: lib/utils/logger.dart [hunk 1]
```diff
@@ -1,15 +1,28 @@
...
-    if (error is ActionCanceledException) {
-      return false;
-    }
-
-    return true;
-  },
+  shouldLogErrorCallback: shouldLogErrorCallback,
 );
+
+final Set<String> _knownBugVersions = {
```
Comment (kunstmusik) [inline lib/utils/logger.dart:12]: Suggest to rename to _knownTransactionDecryptionBugVersions to clarify

File: lib/core/crypto/crypto.dart [hunk 2]
```diff
@@ -127,16 +126,24 @@ class ArDriveCrypto {
 
         decryptedData = Uint8List.fromList(decryptedDataAsListInt);
       } else {
-        throw TransactionDecryptionException();
+        throw UnknowCipherException(
```
Comment (kunstmusik) [inline lib/core/crypto/crypto.dart:129]: Typeo, rename to `UnknownCipherException`

File: packages/ardrive_logger/lib/src/logger.dart [hunk 3]
```diff
@@ -78,6 +78,7 @@ class Logger {
   void e(String message, [Object? error, StackTrace? stackTrace]) {
     String errorMessage = message;
 
+    _shouldLogErrorCallback?.call(error);
```
Comment (kunstmusik) [inline packages/ardrive_logger/lib/src/logger.dart:81]: This doesn't look used correctly, shouldn't the return value be checked?

Comment (thiagocarvalhodev) [reply packages/ardrive_logger/lib/src/logger.dart:81]: Was just for test. Thanks for catching!

**Review by kunstmusik — APPROVED** [review]

**Review by kunstmusik — APPROVED** [review]

**Pull request merged**